### DISCUSSION TITLE
NLP Update error for Text Classifier Spacy v3

### DISCUSSION THREAD

**Discussion by rishkum** [explosion/spaCy#7682] [Help: Coding & Implementations] [answered]
Hi I am getting this error in Spacy 3.+

ValueError: [E989] `nlp.update()` was called with two positional arguments. This may be due to a backwards-incompatible change to the format of the training data in spaCy 3.0 onwards. The 'update' function should now be called with a batch of Example objects, instead of `(text, annotation)` tuples. 
```
import spacy
import random
import json
from spacy.training import Example

from spacy.pipeline.textcat import DEFAULT_SINGLE_TEXTCAT_MODEL
config = {
   "threshold": 0.5,
   "model": DEFAULT_SINGLE_TEXTCAT_MODEL,
}

TRAINING_DATA = [
    ["My little kitty is so special", {"KAT": True}],
    ["Dude, Totally, Yeah, Video Games", {"KAT": False}],
    ["Should I pay $1,000 for the iPhone X?", {"KAT": False}],
    ["The iPhone 8 reviews are here", {"KAT": False}],
    ["Noa is a great cat name.", {"KAT": True}],
    ["We got a new kitten!", {"KAT": True}]
]

nlp = spacy.blank("en")
textcat = nlp.add_pipe("textcat")
textcat.add_label("KAT")

optimizer = nlp.initialize()

# Start the training
nlp.initialize()

# Loop for 10 iterations
for itn in range(100):
    # Shuffle the training data
    random.shuffle(TRAINING_DATA)
    losses = {}

    # Batch the examples and iterate over them
    for batch in spacy.util.minibatch(TRAINING_DATA, size=1):
        texts, annotations = zip(*batch)
        nlp.update(texts, annotations, sgd=optimizer, losses=losses)
    print("Losses", losses)
```

**Comment by polm** [comment, marked as answer]
Did you try calling the update function with a batch of examples instead of (text, annotation) tuples?

Please see the [update documentation](https://spacy.io/api/language#update). Also note that we strongly recommend using the training config with v3 rather than writing your own training loop.

**Reply by adrianeboyd** [reply]
You can also see a demo project with a training config here: https://github.com/explosion/projects/tree/v3/pipelines/textcat_demo
